src/coreclr/jit/llvm.cpp

```diff
@@ -135,31 +142,135 @@ llvm::Type* getLlvmTypeForCorInfoType(CorInfoType corInfoType) {
         case CorInfoType::CORINFO_TYPE_BYREF:
         case CorInfoType::CORINFO_TYPE_CLASS:
-            failFunctionCompilation();
+            return Type::getInt8PtrTy(_llvmContext);
 
         default:
             failFunctionCompilation();
     }
 }
 
+
+unsigned int padOffset(CorInfoType corInfoType, unsigned int atOffset)
+{
+    unsigned int alignment;
+    if (corInfoType == CorInfoType::CORINFO_TYPE_BYREF || corInfoType == CorInfoType::CORINFO_TYPE_CLASS)
+    {
+        // simplified for just pointers
+        alignment = TARGET_POINTER_SIZE; // TODO Wasm64 aligns pointers at 4 or 8?
+    }
+    else
+    {
+        // TODO: value type field alignment - this is the ILToLLVMImporter logic:
+        //var fieldAlignment = type is DefType && type.IsValueType ? ((DefType)type).InstanceFieldAlignment
+        //                                                         : type.Context.Target.LayoutPointerSize;
+        //var alignment      = LayoutInt.Min(fieldAlignment, new LayoutInt(ComputePackingSize(type))).AsInt;
+        //var padding        = (atOffset + (alignment - 1)) & ~(alignment - 1);
+        failFunctionCompilation();
+    }
+    return (atOffset + (alignment - 1)) & ~(alignment - 1);
+}
+
+unsigned int padNextOffset(CorInfoType corInfoType, unsigned int atOffset)
+{
+    unsigned int size;
+    if (corInfoType == CorInfoType::CORINFO_TYPE_BYREF || corInfoType == CorInfoType::CORINFO_TYPE_CLASS)
+    {
+        size = TARGET_POINTER_SIZE;
+    }
+    else
+    {
+        // TODO: value type field size - this is the ILToLLVMImporter logic:
+        // var size = type is DefType && type.IsValueType ? ((DefType)type).InstanceFieldSize
+        //                                               : type.Context.Target.LayoutPointerSize;
+        failFunctionCompilation(); // TODO value type sizes and alignment
+    }
+    return padOffset(corInfoType, atOffset) + size;
+}
+
+/// <summary>
+/// Returns true if the type can be stored on the LLVM stack
+/// instead of the shadow stack in this method.
+/// </summary>
+bool canStoreTypeOnLlvmStack(CorInfoType corInfoType)
+{
+    // TODO: CORINFO_TYPE_VALUECLASS maps to a c# struct?
+    if (corInfoType == CorInfoType::CORINFO_TYPE_VALUECLASS)
+    {
+        // TODO: the equivalent of this c# goes here
+        //if (type is DefType defType)
+        //{
+        //    if (!defType.IsGCPointer && !defType.ContainsGCPointers)
+        //    {
+        //        return true;
+        //    }
+        //}
+        failFunctionCompilation();
+    }
+
+    if (corInfoType == CorInfoType::CORINFO_TYPE_BYREF || corInfoType == CorInfoType::CORINFO_TYPE_CLASS)
+    {
+        return false;
+    }
+    if (corInfoType == CorInfoType::CORINFO_TYPE_REFANY || corInfoType == CorInfoType::CORINFO_TYPE_VAR)
+    {
+        //TODO: what scenarios have these
```

Comment:
`CORINFO_TYPE_VAR` used to be used for verification:
```
// CORINFO_TYPE_VAR is for a generic type variable.
// Generic type variables only appear when the JIT is doing
// verification (not NOT compilation) of generic code
// for the EE, in which case we're running
// the JIT in "import only" mode.
```
Verification is no more, so you should never see it.

`CORINFO_TYPE_REFANY` is a `TypedReference`.

Reply:
Thanks, I'll ignore `CORINFO_TYPE_VAR ` and place `CORINFO_TYPE_REFANY` on the shadow stack.